server/src/helpers/plugins.helper.ts

```diff
@@ -0,0 +1,47 @@
+import { Injectable, InternalServerErrorException } from '@nestjs/common';
+import { decode } from 'js-base64';
+import { requireFromString } from 'module-from-string';
+import { Plugin } from 'src/entities/plugin.entity';
+import { InjectRepository } from '@nestjs/typeorm';
+import { Repository } from 'typeorm';
+import allPlugins from '@tooljet/plugins/dist/server';
+
+@Injectable()
+export class PluginsHelper {
+  private readonly plugins: any = {};
+  private static instance: PluginsHelper;
+
+  constructor(
+    @InjectRepository(Plugin)
+    private pluginsRepository: Repository<Plugin>
+  ) {
+    if (PluginsHelper.instance) {
+      return PluginsHelper.instance;
+    }
+
+    PluginsHelper.instance = this;
+    return PluginsHelper.instance;
+  }
+
+  async getService(pluginId: string, kind: string) {
+    try {
+      if (pluginId) {
+        let decoded: string;
+        if (this.plugins[pluginId]) {
+          decoded = this.plugins[pluginId];
+        } else {
+          const plugin = await this.pluginsRepository.findOne({ where: { id: pluginId } });
+          decoded = decode(plugin.indexFile.data.toString());
```

Comment:
Shouldn't we enforce utf-8 decode here as well?

Reply:
default encoding is `utf-8` for `toString`